src/components/content-tab-assistant.tsx

```diff
@@ -71,62 +77,153 @@ const OfflineModeView = () => {
 	);
 };
 
-const AuthenticatedView = memo(
-	( {
-		messages,
-		isAssistantThinking,
-		updateMessage,
-		siteId,
-	}: {
-		messages: MessageType[];
-		isAssistantThinking: boolean;
-		updateMessage: (
-			id: number,
-			codeBlockContent: string,
-			cliOutput: string,
-			cliStatus: 'success' | 'error',
-			cliTime: string
-		) => void;
-		siteId: string;
-	} ) => {
+type OnUpdateMessageType = (
+	id: number,
+	codeBlockContent: string,
+	cliOutput: string,
+	cliStatus: 'success' | 'error',
+	cliTime: string
+) => void;
+
+interface AuthenticatedViewProps {
+	messages: MessageType[];
+	isAssistantThinking: boolean;
+	updateMessage: OnUpdateMessageType;
+	siteId: string;
+}
+
+export const AuthenticatedView = memo(
+	( { messages, isAssistantThinking, updateMessage, siteId }: AuthenticatedViewProps ) => {
 		const endOfMessagesRef = useRef< HTMLDivElement >( null );
+		const [ showThinking, setShowThinking ] = useState( false );
+		const lastMessage = useMemo(
+			() =>
+				showThinking
+					? ( { role: 'assistant', id: -1, createdAt: Date.now() } as MessageType )
+					: messages[ messages.length - 1 ],
+			[ messages, showThinking ]
+		);
+		const messagesToRender =
+			messages[ messages.length - 1 ]?.role === 'assistant'
+				? messages.slice( 0, -1 )
+				: [ ...messages ];
```

Comment:
I wonder if we need to copy the messages array. I think we are not mutating it, so could we use the original value?
```suggestion
				: messages;
```

Reply:
You are right. Originally, I did that to make react understand that the array was changed. It's not needed now.
Thanks!

Reply:
Fixed